server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -1018,6 +1018,29 @@ public DocTableInfo renameColumn(Reference refToRename, ColumnIdent newName) {
         );
     }
 
+    public static void checkTotalColumnsLimit(RelationName name,
+                                              Settings indexSettings,
+                                              List<Reference> columns) {
+        columns = columns.stream().filter(col -> !col.column().isSystemColumn()).toList();
+        long allowedTotalColumns = TOTAL_COLUMNS_LIMIT.get(indexSettings);
+        if (columns.size() > allowedTotalColumns) {
+            throw new IllegalArgumentException("Limit of total columns [" + allowedTotalColumns + "] in table [" + name + "] exceeded");
+        }
```

Comment:
```suggestion
        long numColumns = columns.stream().filter(col -> !col.column().isSystemColumn()).count();
        long allowedTotalColumns = TOTAL_COLUMNS_LIMIT.get(indexSettings);
        if (numColumns > allowedTotalColumns) {
            throw new IllegalArgumentException("Limit of total columns [" + allowedTotalColumns + "] in table [" + name + "] exceeded");
        }
```

No need to materialize a list to get the updated count